File: leetcode/0387. First Unique Character in a String/Solution.swift
// https://leetcode.com/problems/first-unique-character-in-a-string/

class Solution {
    func firstUniqChar(_ s: String) -> Int {
        var counts = [String: Int]()
        for c in s {
            let cs = String(c)
            counts[cs] = (counts[cs] ?? 0) + 1
        }
        
        for (i, c) in s.enumerated() {
            let cs = String(c)
            if (counts[cs] ?? 0) == 1 {
                return i
            }
        }
        
        
        return -1
    }
}

File: leetcode/0387. First Unique Character in a String/Solution.py
# https://leetcode.com/problems/first-unique-character-in-a-string/

import collections

class Solution:
    def firstUniqChar(self, s: str) -> int:
        counts = collections.defaultdict(int)
        for char in list(s):
            counts[char] += 1
        
        for i, char in enumerate(s):
            if counts[char] == 1:
                return i

        return -1

s = Solution()
assert s.firstUniqChar("leetcode") == 0
assert s.firstUniqChar("loveleetcode") == 2